pkg/util/podstore/podstore.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package podstore
+
+import (
+	"fmt"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+)
+
+const (
+	DeleteQueueName = "pod_delete"
+	PodIPIndex      = "PodIP"
+	DelayTime       = time.Minute * 5
+)
+
+// for testing
+var delayTime = DelayTime
+
+type PodStorage struct {
+	curPod  cache.Indexer
+	prevPod cache.Indexer
+	// Could be other lighter implementations
```

Comment:
I've removed it. I also add a benchmark for func `BenchmarkGetPodByIPAndTime`. Do you think it is sufficient if I only test this function?

Reply:
`GetPodByIPAndTime` does not use the `workqueue`, so that benchmark doesn't seem relevant to the choice of `workqueue.DelayingInterface`.
If we wanted a valid benchmark, it would have to generate a lot of Pod Add & Delete events.

Reply:
Do you mean we should generate a lot of Pod Add & Delete events in `BenchmarkGetPodByIPAndTime `? I'm not quite sure how to test `workqueue.DelayingInterface` as currently it's only called when we have Delete events. And do you know how many events we should generate based on the number of Pods?

Reply:
you could create 1,000 Pods in a fake apiserver, then delete all of them at once.

Reply:
I've updated the benchmark test. But I'm still thinking that even I generate lots of Pod Add/Delete events, I still can't tell whether `workqueue.DelayingInterface` plays a role here? When we have Delete events, the `workqueue` is only used by `s.podsToDelete.AddAfter(pod, DelayTime)` in the DeleteFunc. Is that because the `workqueue` can occupy some resource when we have Delete events?
Or should I have another benchmark for `workqueue.DelayingInterface`?